lua/entities/gmod_wire_expression2/core/cl_http_cl.lua

```diff
@@ -0,0 +1,129 @@
+--[[
+
+	Clientside HTTP library
+	 - swadical 🍌
+
+]]
+
+--	Clientside HTTP is separate from serverside HTTP:
+--		Serverside HTTP allows greater speeds than clientside HTTP and it is
+--		up to the server owner to decide which extension should be enabled
+--		through a convar (found in core/http.lua)
+
+--	This library will mirror the functionality of core/http.lua as best as
+--	possible, making sure any existing e2s which make use of the http library
+--	are not broken by this extension
+
+E2Lib.clHTTP = {}
+local lib = E2Lib.clHTTP
+
+lib.netMsgID = "wire_expression2_cl_http" -- open to change
+
+-- string function(void)
+function lib:newUID()
+	return ("0x%08xf"):format(math.random(1,0xD4A51000))-- return 0x(hex) between 0 and 10^12
```

Comment:
I would've just used a number that incremented for each new request, and reset if it got above a certain amount. For example 255. Then you could use `net.WriteUInt( uid, 8 )` instead of `net.WriteString( *10 char string* )` which would save you some bytes